background stuff to make << and >> go away

Spookerton opened the pull request:
is split into commits for review purposes again, hurray.
replaces all i/o with macros, cleans up existing duplicate macros, adds semantic macros for savefile access

will hopefully minimize future << conflicts

Spookerton committed:
comments, user << message, tag terminators, hrefs

thasc commented on code/__defines/integrated_circuits.dm:
```diff
@@ -11,9 +11,9 @@
-#define IC_ACTION_LONG_RANGE	(1<<2) // If the circuit communicate with something outside of the assembly
+#define IC_ACTION_MOVEMENT    0x1 // If the circuit can move the assembly
+#define IC_ACTION_COMBAT      0x2 // If the circuit can cause harm
+#define IC_ACTION_LONG_RANGE  0x3 // If the circuit communicate with something outside of the assembly
```
Should this be 0x4?

Spookerton replied:
Oh, so it should be.

Lohikar commented on code/_helpers/sorts/__main.dm:
```diff
@@ -169,7 +169,7 @@ reverse a descending sequence without violating stability.
 	while(n >= MIN_MERGE)
 		r |= (n & 1)
-		n >>= 1
+		n = n >> 1
```
Artifact from the old changes, looks like.

Lohikar commented on code/_macros.dm:
```diff
@@ -99,17 +99,23 @@
-#define send_rsc(target, rsc_content, rsc_name)             target << browse_rsc(rsc_content, rsc_name)
-#define open_link(target, url)             target << link(url)
+/// General I/O helpers
+#define to_target(target, payload)            target << (payload)
```
This seems kind've overly generic? A quick C-f shows it's only used in one spot other than this file.

Spookerton replied:
As conversation -- Generic, but so that the likelihood of conflicts is minimized since I'm no longer removing the test.

Lohikar commented on code/modules/admin/NewBan.dm:
```diff
@@ -114,23 +114,23 @@ var/savefile/Banlist
-		Banlist["reason"] << reason
-		Banlist["bannedby"] << bannedby
-		Banlist["temp"] << temp
+		to_file(Banlist["key"], ckey)
```
This is `/savefile` ops afaik, not file - should probably be left as `<<`?

Lohikar commented on code/modules/admin/admin.dm:
```diff
@@ -270,7 +270,7 @@ var/global/floorIsLava = 0
 	var/savefile/S=new("data/player_notes.sav")
 	var/list/note_keys
-	S >> note_keys
+	from_file(S, note_keys)
```
Also savefile ops.

Lohikar commented on code/modules/admin/banjob.dm:
```diff
@@ -46,9 +46,9 @@ var/jobban_keylist[0]		//to store the keys & ranks
 	if(config.ban_legacy_system)
 		var/savefile/S=new("data/job_full.ban")
-		S["keys[0]"] >> jobban_keylist
+		from_file(S["keys[0]"],  jobban_keylist)
```
Also savefile ops.

Lohikar commented on code/modules/admin/player_notes.dm:
```diff
@@ -72,7 +8,7 @@ datum/admins/proc/notes_gethtml(var/ckey)
 	var/savefile/info = new("data/player_saves/[copytext(key, 1, 2)]/[key]/info.sav")
 	var/list/infos
-	info >> infos
+	from_file(info, infos)
```
More savefile ops in this file, not flagging each one.

Lohikar commented on code/modules/admin/topic.dm:
```diff
@@ -172,7 +172,7 @@
 			if(!D)	return
 			var/list/permissionlist = list()
-			for(var/i=1, i<=R_MAXPERMISSION, i<<=1)		//that <<= is shorthand for i = i << 1. Which is a left bitshift
+			for(var/i=1, i<=R_MAXPERMISSION, i = i << 1)
```
Another artifact of the bitshift macro.

Lohikar commented on code/modules/admin/topic.dm:
```diff
@@ -308,10 +308,10 @@
-		Banlist["temp"] << temp
-		Banlist["minutes"] << minutes
-		Banlist["bannedby"] << usr.ckey
+		to_file(Banlist["reason"], reason)
```
More savefiles.

Lohikar commented on code/modules/admin/view_variables/helpers.dm:
```diff
@@ -8,16 +8,16 @@
 		<a href='?_src_=vars;datumedit=\ref[src];varnameedit=name'><b>[src]</b></a>
 		<br><font size='1'>
-		<a href='?_src_=vars;rotatedatum=\ref[src];rotatedir=left'><<</a>
+		<a href='?_src_=vars;rotatedatum=\ref[src];rotatedir=left'><=</a>
```
The changes in this file can probably be reverted, unless this is also a stylistic change.

Spookerton replied:
It is!

Lohikar commented on code/modules/client/preference_setup/general/04_flavor.dm:
```diff
@@ -7,35 +7,35 @@
-	S["flavor_texts_hands"]		>> pref.flavor_texts["hands"]
-	S["flavor_texts_legs"]		>> pref.flavor_texts["legs"]
-	S["flavor_texts_feet"]		>> pref.flavor_texts["feet"]
+	from_file(S["flavor_texts_general"], pref.flavor_texts["general"])
```
Save files.

Lohikar commented on code/modules/client/preference_setup/global/01_ui.dm:
```diff
@@ -11,18 +11,18 @@
-	S["UI_style_alpha"]	>> pref.UI_style_alpha
-	S["ooccolor"]		>> pref.ooccolor
-	S["clientfps"]		>> pref.clientfps
+	from_file(S["UI_style"], pref.UI_style)
```
Save files.

Lohikar commented on code/modules/client/preference_setup/global/04_ooc.dm:
```diff
@@ -7,11 +7,11 @@
 
 /datum/category_item/player_setup_item/player_global/ooc/load_preferences(var/savefile/S)
-	S["ignored_players"]	>> pref.ignored_players
+	from_file(S["ignored_players"], pref.ignored_players)
```
Save files.

Lohikar commented on code/modules/client/preference_setup/matchmaking/relations.dm:
```diff
@@ -7,12 +7,12 @@
 /datum/category_item/player_setup_item/relations/load_character(var/savefile/S)
-	S["relations"]	>> pref.relations
-	S["relations_info"]	>> pref.relations_info
+	from_file(S["relations"], pref.relations)
```
Save files.

Lohikar commented on code/modules/client/preference_setup/loadout/loadout.dm:
```diff
@@ -127,7 +127,7 @@ var/list/gear_datums = list()
 	. += "<table align = 'center' width = 100%>"
 	. += "<tr><td colspan=3><center>"
-	. += "<a href='?src=\ref[src];prev_slot=1'>\<\<</a><b><font color = '[fcolor]'>\[[pref.gear_slot]\]</font> </b><a href='?src=\ref[src];next_slot=1'>\>\></a>"
+	. += "<a href='?src=\ref[src];prev_slot=1'>\<=</a><b><font color = '[fcolor]'>\[[pref.gear_slot]\]</font> </b><a href='?src=\ref[src];next_slot=1'>=\></a>"
```
Stylistic?

Spookerton replied:
Yes!

Lohikar commented on code/modules/client/preferences.dm:
```diff
@@ -321,7 +321,7 @@ datum/preferences
 		for(var/i=1, i<= config.character_slots, i++)
 			S.cd = GLOB.using_map.character_load_path(S, i)
-			S["real_name"] >> name
+			from_file(S["real_name"], name)
```
Savefiles.

Lohikar commented on code/modules/client/preferences_savefile.dm:
```diff
@@ -13,7 +13,7 @@
 	S.cd = "/"
 
-	S["version"] >> savefile_version
+	from_file(S["version"], savefile_version)
```
Savefiles.

Lohikar commented on code/modules/hydroponics/spreading/spreading.dm:
```diff
@@ -184,7 +184,7 @@
 	var/list/dirList = list()
 
-	for(var/i=1,i<=16,i <<= 1)
+	for(var/i=1,i<=16, i = i << 1)
```
Another macro artifact.

Lohikar commented on code/modules/maps/swapmaps.dm:
```diff
@@ -219,11 +219,11 @@ swapmap
-		S["y"] << y2-y1+1
-		S["x"] << x2-x1+1
-		S["areas"] << areas
+		to_file(S["id"], id)
```
Savefiles.

Lohikar commented on code/_macros.dm:
```diff
@@ -99,17 +99,23 @@
+#define send_link(target, url)                to_target(target, link(url))
+#define send_output(target, msg, control)     to_target(target, output(msg, control))
+#define to_file(file_handle, source_var)      to_target(file_handle, source_var)
+#define from_file(file_handle, target_var)    from_target(file_handle, target_var)
```
I don't think there's any legitimate use of this macro, this syntax is only valid on savefiles and they shouldn't be using `*_file` macros imo - should get their own set, if anything.

Lohikar reviewed (commented):
Savefiles should be treated separately from files imo, + there's some changes left over from the macro.

Spookerton committed:
cleaner io macros + non-savefile uses

image_to, sound_to, links, browser, rsc, output, sending logs, writing logs
showing ai and robots at round end cleanup to not break

Spookerton committed:
to/from savefiles

Spookerton committed:
update checks

CrimsonShrike commented on code/modules/mob/animations.dm:
```diff
@@ -105,7 +105,7 @@ note dizziness decrements automatically in the mob's Life() proc.
 
 	var/amplitude = 2 //maximum displacement from original position
-	var/period = 36 //time taken for the mob to go up >> down >> original position, in deciseconds. Should be multiple of 4
+	var/period = 36 //time taken for the mob to go up -> down -> original position, in deciseconds. Should be multiple of 4
```
Is this necessary too?

Spookerton replied:
Pretty much.

CrimsonShrike reviewed (approved):
Seems correct. Although mass replace PRs hurt me physically so could only look for the briefest of moments

NullSnapshot commented on code/_macros.dm:
```diff
@@ -99,17 +99,24 @@
+#define to_world(message)                     to_target(world, message)
+#define to_world_log(message)                 to_target(world.log, message)
+#define to_chat(target, message)              to_target(target, message)
+#define sound_to(target, sound)               to_target(target, sound)
```
Just a semantic thing, but I feel like it'd make more sense if sound_to and image_to were changed to send_sound and send_image respectively. It'd make them semantically similar to send_rsc, send_link, and send_output.

Spookerton replied:
I'd actually like to, but sound_to has a large set of existing uses and this PR touches a lot of files already. I chose to go with image_to alike to sound_to because it was the drastically less used one. Perhaps in a second pass where it'll be less noisy.

NullSnapshot replied:
sounds good to me then!

NullSnapshot reviewed (approved):
Apart from my one comment, this all looks fine to me.

Spookerton commented:
Witness.
